cmd/prepare-storage.go

```diff
@@ -190,7 +178,14 @@ func prepForInitXL(firstDisk bool, sErrs []error, diskCount int) InitActions {
 		return WaitForHeal
 	}
 
-	// Exhausted all our checks, un-handled errors perhaps we Abort.
+	// Some disks unformatted, Some disks formatted and some disks are offline.
+	// This is an undecisive state - wait for all of offline disks to be online
+	// to figure out the course of action.
+	if disksUnformatted+disksFormatted+disksOffline == diskCount {
+		return WaitForAll
```

Comment:
Here we should return `WaitForHeal`. I.e we can initialize object layer when we have `readQuorum` number of formatted disks.

Reply:
Explained on why its already handled..

Reply:
I am not sure, isn't this an infinite waiting ? shouldn't we able to run the cluster even if we have some nodes < quorum offline ?

Reply:
We should wait regardless since we don't have enough quorum to decide anything. @vadmeste . If there are enough corrupted we exit anyways. It is when there are disks offline and we don't have quorum number of formatted disks and some look unformatted. We have to wait infinitely you have to look at the entire flow to understand the state machine here @vadmeste .